File: Heat_Recognition_System/frontend/src/api/dietLog.ts
import request from '@/utils/request'

export interface DietLog {
  id?: number
  userId?: number
  date: string
  mealType?: string
  totalCalories: number
  protein: number
  fat: number
  carbs: number
  createTime?: string
}

export const getDietLogAPI = (date: string) =>
  request<DietLog | null>({
    url: '/diet-log',
    method: 'get',
    params: { date }
  })

export const getDietHistoryAPI = () =>
  request<DietLog[]>({
    url: '/diet-log/history',
    method: 'get'
  })

File: Heat_Recognition_System/backend/pojo/src/main/java/com/cy/entity/DietLog.java
package com.cy.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 饮食日志（每日/每餐汇总）
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DietLog {

    private Long id;

    private Long userId;

    /** 日期 */
    private LocalDate date;

    /** 餐次 */
//    private String mealType;

    /** 营养汇总 */
    private Double totalCalories;
    private Double protein;
    private Double fat;
    private Double carbs;

    private LocalDateTime createTime;
}